### DISCUSSION TITLE
Disable syntax highlighting completely?

### DISCUSSION THREAD

**Discussion by 0xdeadbeer** [nvim-treesitter/nvim-treesitter#4033] [Q&A] [answered]
For the last day I've been banging my head against a wall trying to fix this issue

![image](https://user-images.githubusercontent.com/64986162/209484453-e5a9ec52-170f-4a6f-b271-60ce04284b51.png)

I see solutions all around the internet that go like.. actually telling nvim where the library is located and so on. But what if I'm lazy and want to simply disable syntax errors tree-sitter shows me? :)

I know there was an issue going on: #119
But all of the solutions provided in that issue didn't work in my case..

 ```
require'nvim-treesitter.configs'.setup {
  ensure_installed = { "c", "cpp", "lua", "rust" },
  sync_install = false,
  auto_install = true,
  ignore_install = { "javascript" },

  highlight = {
    enable = true,
    disable = { },
    additional_vim_regex_highlighting = false,
  }
}

require "nvim-treesitter.highlight"
local hlmap = vim.treesitter.TSHighlighter.hl_map

hlmap.error = nil
```
This will cause the following error on restart: 
![image](https://user-images.githubusercontent.com/64986162/209484525-c5ae87a6-602c-46bd-9876-49aa9275d52b.png)
<br>
The following solution also didn't work: (if i enter it into my nvim terminal)
```
highlight link TSError Normal
```
It doesn't change anything?

I'm really confused and would really appreciate it if someone could point me in the right direction please
It's such a simple problem yet so difficult to solve at first..

**Comment by theHamsta** [comment, marked as answer]
> local hlmap = vim.treesitter.TSHighlighter.hl_map
> hlmap.error = nil

`hlmap` does not exist anymore. The hlmap used to map `@error` to `TSError`. You could do with `TSError` whatever you want (it was never mapped to nothing by nvim-tresitter. Now you can map `@error` directly. Neovim does not map `@error` to anything. Maybe it is your colorscheme?. You can map `@error` to anything. If you link `@error` to something non-existing `@highlight link @error doesnotexist` you should not see it. `highlight link @error Normal` should also work.


The missing import is a LSP error, not a tree-sitter syntax error. It has a different highlight group: read `:h hl-DiagnosticError`. Tree-sitter only understands the syntax, not it's semantics. It does not understand what `#include` means and therefore also doesn't cause the error about the missing import.

To get rid of the missing imports use CMake's `-DCMAKE_EXPORT_COMPILE_COMMANDS=ON` and symlink `compile-commands.json` to the root dir. Meson generates `compile-commands.json` automatically. For everything else, (e.g. makefiles). Use `bear -- make` instead of `make` (https://github.com/rizsotto/Bear, `apt install bear`). It will create `compile-commands.json` and you're fine.
